src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -101,6 +101,9 @@ public abstract class Connection : IDisposable
 
             protected Connection()
             {
+#if DEBUG
+                _creationCallStack = Environment.StackTrace;
```

Comment:
why only do this in debug?  we create connections rarely right?  so this should be cheap enough to do even in release.

Reply:
no connection is created many times. and we see this issue again we can make this release later.